wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceRuntimes.java

```diff
@@ -785,10 +793,16 @@ private boolean containsThisRuntimesId(String workspaceId) {
     return environmentFactories.keySet();
   }
 
-  private InternalEnvironment createInternalEnvironment(
-      Environment environment, Map<String, String> workspaceConfigAttributes)
+  @VisibleForTesting
+  InternalEnvironment createInternalEnvironment(
+      @Nullable Environment environment, Map<String, String> workspaceConfigAttributes)
       throws InfrastructureException, ValidationException, NotFoundException {
-    String recipeType = environment.getRecipe().getType();
+    String recipeType;
+    if (environment == null) {
+      recipeType = Constants.NO_ENVIRONMENT_RECIPE_TYPE;
```

Comment:
I like the idea but we use `no-environment` recipe returned in the list of supported environments to allow clients (aka UD) to use it properly. Because of that, keeping it as one of the environment implementations simplifies the code in my opinion. I would leave it like that for the time being.

Reply:
Good point with UD, I missed it before. I'm OK with the current approach since I can't propose an easy alternative way which would expose supporting of `no-environment` for clients.